### ISSUE TITLE
package status?

### ISSUE THREAD

**Issue by jwaldmann** [d-day/relation#1]
Hi. 

Are there active users for this package? Active developers? (It seems ghc-7.8 cannot compile it, but this is trivial to fix by removing or bumping version bounds in the cabal file.)

I might be interested in "taking over" this, because I have some applications, and use a similar package
with the same basic idea (represent relation by maps in both directions). I need some extra features, see  http://autolat.imn.htwk-leipzig.de/gitweb/?p=autolib;a=tree;f=relation/Autolib/Relation;hb=HEAD 

The main incompatibility is probably that I want to store domain and range (as sets) in the relation object. Mathematically, I want a relation to be a triple (D,R,T) where D = domain, T = target, and R \subseteq D \times T. 

"relation" is quite a general name for the package. Ultimately, it should be part of "containers"?
- J.

**Comment by no-identd** [comment]
You might want to check out #3

**Comment by newhoggy** [comment]
Is this package maintained?

**Comment by newhoggy** [comment]
@d-day I'm happy to take over maintenance of this package and make it compile with the latest versions of GHC, etc.

**Comment by d-day** [comment]
John: By all means. You have my permission. If you need me to do anything
else, please let me know.

**Comment by jwaldmann** [comment]
Update to my ancient message above: my relation code is now at
https://gitlab.imn.htwk-leipzig.de/autotool/all0/tree/master/relation/Autolib/Relation

One next step is generalizing to functions (represented as sparse matrices) with values in arbitrary semirings (not just Bool), cf. https://gitlab.imn.htwk-leipzig.de/waldmann/pure-matchbox/blob/master/src/Data/Relation/Weighted.hs

**Comment by no-identd** [comment]
Cc @JeffreyBenjaminBrown

**Comment by no-identd** [comment]
Also Cc @dmwit @technogeeky

**Comment by newhoggy** [comment]
Thanks @d-day.

Please can add haskellworks as a package maintainer.

I will fork the package and tidy up the project to build with CI and the latest GHC and then look at patches.

**Comment by no-identd** [comment]
@newhoggy I assume you mean @haskell-works? (https://haskell-works.github.io/ ?)

**Comment by newhoggy** [comment]
Yep, except on hackage there is no dash: https://hackage.haskell.org/user/haskellworks

**Comment by newhoggy** [comment]
Hello, any update on adding myself as a package maintainer?

**Comment by newhoggy** [comment]
Hello @d-day , any update on adding myself as a package maintainer?

**Comment by d-day** [comment]
I will take a look tonight after work. I hope I can find the appropriate
credentials. Can you specify what exactly you want done: do you want
something done on hackage, on github, or both?

Thanks,
-Drew

**Comment by newhoggy** [comment]
Visit this page:

https://hackage.haskell.org/package/relation/maintainers/

Click [Edit] if you're not logged on already.

Add haskellworks as a maintainer.

Thanks!

**Comment by d-day** [comment]
Mr. Ky:

I have completed this. Let me know if you need anything else from me. Sorry
for the delay.

**Comment by newhoggy** [comment]
Thanks!

**Comment by newhoggy** [comment]
I'll be developing the project from this fork: https://github.com/haskell-works/relation.  Please file tickets and new PRs against the new fork.  Thanks!
